sips/instance_decided_enforcement.md

```diff
@@ -0,0 +1,150 @@
+| Author      | Title                        | Category | Status |
+|-------------|------------------------------|----------|--------|
+| Alon Muroch | Instance Decided Enforcement | Core     | open-for-discussion  |
+
+**Summary**
+
+No longer require a QBFT instance to decide in order to start the next one.
+If an instance is stalled, it will be terminated after a certain round or when a new instance starts.
+The instance's height will be the duty's slot number. This will allow for easier syncing.
+
+**Rationale**
+
+In our current implementation, a QBFT instance can only start if the previous one has decided.
+This is done to ensure that all nodes are synchronized and that the instance height is predictable.
+This is not required since we can use slots to sync decided instances and the instance's height can be the duty's slot number.
+
+The change will allow for more efficient constant timeouts without risking the liveness of the protocol. Currently, there is a liveness risk with constant timeouts.
+F + 1 nodes that stall behind the committee's rounds can delay consensus indefinitely and prevent the start of new instances.
+Since after 2 epochs, the duty will expire and will not be able to gain rewards, we can terminate the instance after a certain round.
+If a new instance starts, the previous one will be terminated. This is to reduce slashing risks and to save on resources. 
+
+**Spec Changes** 
+
+**Timeout**
+~~~go
+var (
+	quickTimeoutThreshold = Round(8)
+	quickTimeout          = 2 * time.Second
+	slowTimeout           = 2 * time.Minute 
+	// CutoffRound which round the instance should stop its timer and progress no further 
+	CutoffRound = 15 // stop processing instances after 8*2+120*6 = 14.2 min (~ 2 epochs)
+)
+
+~~~
+
+**QBFT Controller**
+- Remove function CanStartInstance
+- Remove function canStartInstanceForValue
+- Remove function bumpHeight
+- Remove future message processing - no need as each instance is independent. If a future decided message is received it will trigger decided sync
+```go
+// StartNewInstance will start a new QBFT instance, if can't will return error
+func (c *Controller) StartNewInstance(height Height, value []byte) error {
```

Comment:
If height is just slot now, maybe we should just call it slot?

Reply:
this will be a huge change for the code and internal states 😅 
also @alonmuroch wanted to keep it as "height"... maybe to keep our concepts seperate from eth

you do have a point though